src/formats.ts

```diff
@@ -0,0 +1,14 @@
+import { format } from "d3-format";
+import { utcFormat, utcParse } from "d3-time-format";
+
+export const yearMonthDayFormat = utcFormat("%Y-%m-%d");
+
+export const formatWithCommas = format(",");
+
+const formatSIPrefix = format(".2s");
+export const formatSIDropTrailingZeroes = (d: number): string =>
+  formatSIPrefix(d).replace(/\.0+/, "");
+
+export const formatAsPercent = format(".0%");
+
+export const yearMonthDayParse = utcParse("%Y-%m-%d") as (s: string) => Date;
```

Comment:
I think this file is arguably net-new, unsure if it's useful? there were some similar declarations in a few places.. maybe not worth the DRY? 🤷

Reply:
I like the idea of consolidating these, though a bit wary of ad-hoc files. I think it's probably fine, but maybe also consider:

- Could some / all of these be implemented as Preact components? e.g. `const Percent = ({ number }) => format(".0%");`
- Maybe [NPM workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces), to give some structure to ad-hocness?
- Constants / configuration file including desired formats?

Reply:
I think some of them could be components for things like the table, but we use the functions directly in a lot of places so we'd still need to expose them?